tests/test/org/commcare/util/test/CaseAPITests.java

```diff
@@ -0,0 +1,123 @@
+
+    private Ledger l;
+
+    private String owner;
+    private String groupOwner;
+    private Vector<String> groupOwned;
+    private Vector<String> userOwned;
+
+
+    @Before
+    public void setUp() throws Exception {
+
+        owner = "owner";
+        groupOwner = "groupowned";
+
+        userOwned = new Vector<String>();
+        userOwned.addElement(owner);
+
+        groupOwned = new Vector<String>();
+        groupOwned.addElement(owner);
+        groupOwned.addElement(groupOwner);
+
+        a = new Case("123", "a");
+        a.setCaseId("a");
+        a.setUserId(owner);
+        a.setID(12345);
+        b = new Case("b", "b");
+        b.setCaseId("b");
+        b.setUserId(owner);
+        c = new Case("c", "c");
+        c.setCaseId("c");
+        c.setUserId(owner);
+        d = new Case("d", "d");
+        d.setCaseId("d");
+        d.setUserId(owner);
+        e = new Case("e", "e");
+        e.setCaseId("e");
+        e.setUserId(groupOwner);
+
+        l = new Ledger("ledger_entity_id");
+        l.setID(12345);
+        l.setEntry("test_section_id", "test_entry_id", 2345);
+    }
+
+    @Test
+    public void testOwnerPurge() {
+        Connection connection = null;
+        PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+        String databaseName = ("test.db");
+        try {
+            connection = DriverManager.getConnection("jdbc:sqlite:" + databaseName);
+
+            SqlHelper.dropTable(connection, "TFLedger");
+
+            SqlHelper.createTable(connection, "TFLedger", new Ledger());
+
+            SqlHelper.insertToTable(connection, "TFLedger", l);
+
+            preparedStatement = SqlHelper.prepareTableSelectStatement(connection, "TFLedger", new String[]{"entity-id"},
+                    new String[]{"ledger_entity_id"}, new Ledger());
+            if (preparedStatement == null) {
+                fail("failed to prepare table select query");
+            }
+            resultSet = preparedStatement.executeQuery();
+            byte[] caseBytes = resultSet.getBytes("commcare_sql_record");
+            DataInputStream is = new DataInputStream(new ByteArrayInputStream(caseBytes));
+
+            Ledger readLedger = new Ledger();
+            PrototypeFactory lPrototypeFactory = new PrototypeFactory();
+            lPrototypeFactory.addClass(Ledger.class);
+            readLedger.readExternal(is, lPrototypeFactory);
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            fail("Unexpected exception " + e.getMessage());
+        } finally{
+            try {
+                if(preparedStatement != null) {
+                    preparedStatement.close();
+                }
+                if(connection != null) {
+                    connection.close();
+                }
+                if(resultSet != null){
+                    resultSet.close();
+                }
+                File databaseFile = new File(databaseName);
```

Comment:
Isn't there an @after annotation you can use to make sure things get cleaned out? Not sure if it's more reliable than finally, but might make the process of

Setting up a db connector, etc

then

closing all connections
removing the file

a bit cleaner.